tests/integration/utils/rest_client.py

```diff
@@ -0,0 +1,78 @@
+"""
+karapace - Test rest client with retries
+
+Copyright (c) 2024 Aiven Ltd
+See LICENSE for details
+"""
+from __future__ import annotations
+
+from aiohttp import BasicAuth
+from collections.abc import Mapping
+from karapace.client import Client, Headers, Path, Result
+from karapace.typing import JsonData
+from tenacity import retry, stop_after_attempt
+
+
+class UnexpectedResponseStatus(Exception):
+    pass
+
+
+class RetryRestClient:
+    def __init__(self, client: Client):
+        self._karapace_client = client
+
+    @retry(stop=stop_after_attempt(5))
```

Comment:
Should we maybe wait a bit between retries?

Reply:
Done.